packages/modules/workflow-engine-inmemory/src/services/workflows-module.ts

```diff
@@ -155,4 +172,12 @@ export class WorkflowsModuleService<
   ) {
     return this.workflowOrchestratorService_.unsubscribe(args as any, context)
   }
+
+  private async clearExpiredExecutions() {
+    return this.manager_.execute(`
+      DELETE FROM workflow_execution
+      WHERE retention_time IS NOT NULL AND
+      updated_at <= (CURRENT_TIMESTAMP - INTERVAL '1 second' * retention_time);
```

Comment:
q: is the updated_at here correct?

Reply:
yeah, this is correct.
it would be the same as 
`(updated_at + INTERVAL '1 second' * retention_time) <= CURRENT_TIMESTAMP`

Reply:
So this would be if a workflow execution has a 5-day retention period and was last updated 6 days ago, it would be deleted. Right?